tiledb/sm/query/query.h

```diff
@@ -340,6 +340,9 @@ class Query {
   Status get_written_fragment_timestamp_range(
       uint32_t idx, uint64_t* t1, uint64_t* t2) const;
 
+  /** Returns the array's smart pointer. */
+  shared_ptr<Array> array_shared();
```

Comment:
I'd recommend `inline` for this function, since all it's doing is returning a member variable.